Model/Sync/Metadata/Processor.php

```diff
@@ -67,55 +71,53 @@ public function __construct(
                             $storeId,
                             $this->yotpoConfig->getStoreName($storeId)
                         )
                     );
+                    continue;
+                }
+                $this->logger->info(
+                    __(
+                        'Starting updating Metadata for Magento Store ID: %1, Name: %2',
+                        $storeId,
+                        $this->yotpoConfig->getStoreName($storeId)
+                    )
+                );
 
-                    $data = $this->prepareMetadata();
-                    $data['entityLog'] = 'general';
-                    $endPoint = $this->yotpoConfig->getEndpoint('metadata');
-                    $response = $this->yotpoSyncMain->syncV1('POST', $endPoint, $data);
-                    if ($response['is_success']) {
-                        $this->logger->info(
-                            __(
-                                'Updating metadata for Magento Store ID: %1, Name: %2 [SUCCESS]',
-                                $storeId,
-                                $this->yotpoConfig->getStoreName($storeId)
-                            )
-                        );
-                    }
-                } catch (\Exception $e) {
+                $metadataDataToSync = $this->prepareMetadata();
+                $metadataDataToSync[$this::ENTITY_LOG_FILE] = 'general';
+                $metadataEndpoint = $this->yotpoConfig->getEndpoint('metadata');
+                $response = $this->yotpoSyncMain->syncV1($this::POST_METHOD_STRING, $metadataEndpoint, $metadataDataToSync);
+                if ($response[$this::SYNC_RESPONSE_IS_SUCCESS_KEY]) {
                     $this->logger->info(
                         __(
-                            'Exception on Magento Store ID: %1, Name: %2, Reason: %3',
+                            'Finished updating Metadata successfully for Magento Store ID: %1, Name: %2',
                             $storeId,
-                            $this->yotpoConfig->getStoreName($storeId),
-                            $e->getMessage()
+                            $this->yotpoConfig->getStoreName($storeId)
                         )
                     );
                 }
+            } catch (\Exception $exception) {
+                $this->logger->info(
+                    __(
+                        'Error occurred when updating Metadata, got Exception on Magento Store ID: %1, Name: %2, Reason: %3',
+                        $storeId,
+                        $this->yotpoConfig->getStoreName($storeId),
+                        $exception->getMessage()
+                    )
+                );
+            } finally {
                 $this->stopEnvironmentEmulation();
             }
-        } catch (\Exception $e) {
-            $this->logger->info(__('Exception: %1', $e->getMessage()));
+            $this->stopEnvironmentEmulation();
```

Comment:
I think this is redundant now because of the added _finally_ clause, WDYT?

Reply:
I think it shouldn't be in finally, because logically its incorrect, catch is used for doing post action after initial action was failed, finally is used for closing open third party connections / threads or recover side effects that occurred, like the environmentEmulation.
Adding the `stopEnvironmentEmulation` in the finally clause was a great idea, but we should keep the log entry in the catch clause.

Reply:
I think it shouldn't be in finally, because logically its incorrect, catch is used for doing post action after initial action was failed, finally is used for closing open third party connections / threads or recover side effects that occurred, like the environmentEmulation.
Adding the stopEnvironmentEmulation in the finally clause was a great idea, but the log entry should remain in the catch clause.

Reply:
I was not referring to the log entry, I was referring to the stopEnvironmentEmulation outside of the _finally_ clause.

![image](https://user-images.githubusercontent.com/45664120/157032764-1fb2cf33-4e15-4045-860b-19a703e37267.png)

Reply:
Oh, ops didn't notice.
But its not redundant, its called after success, so for next store it will start environment emulation again.

Reply:
Finally will be called after a successful execution as well, and not only after an exception was raised. I'm not sure why do we need it there.

Reply:
Right, removed it, nice catch